apps/server/src/modules/lesson/repo/lesson.repo.ts

```diff
@@ -65,4 +65,16 @@ export class LessonRepo extends BaseRepo<LessonEntity> {
 
 		return lessons;
 	}
+
+	public async removeUserReference(userId: EntityId): Promise<number> {
+		const id = new ObjectId(userId);
+		// replace all user references with undefined
+		const count = await this._em.nativeUpdate(
+			this.entityName,
+			{ contents: { $elemMatch: { user: id } } } as FilterQuery<LessonEntity>,
+			{ $set: { 'contents.$.user': undefined } } as Partial<LessonEntity>
```

Comment:
Did you think about removing the property altogether (because it is optional) instead of setting it to `undefined`? What would be better? Or doesn't it matter?

Reply:
probably same thing 
syntax should look like `{ $unset: { 'contents.$.user': '' } }` which is not necesarly easier to read
But not sure if undefined which would make the field null in database is the same as delete... I am inclined to favor having in db a null valued field, than not having the field at all

Reply:
It should be spend time to figure out which it the most robust implementation. 
For reading the $unset description it look like we should use it, instead of $set undefined.
But i do not look deeper into it.

Reply:
I changed to $unset. This makes sure we have no null data in the collection.